core/jobs/batch_jobs/exp_proto_size_calculation_job.py

```diff
@@ -0,0 +1,424 @@
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+    from mypy_imports import exp_models
+    from mypy_imports import opportunity_models
+
+(base_models, exp_models, opportunity_models) = (
+    models.Registry.import_models(
+        [models.Names.BASE_MODEL, models.Names.EXPLORATION,
+         models.Names.OPPORTUNITY]))
+datastore_services = models.Registry.import_datastore_services()
+
+
+class PopulateExplorationProtoSizeInBytesJob(base_jobs.JobBase):
+    """Job that populates android_proto_size_in_bytes attribute
+    for Exploration models."""
+
+    @staticmethod
+    def _migrate_exploration(
+        exp_id: str,
+        exp_model: exp_models.ExplorationModel
+    ) -> result.Result[
+        Tuple[str, exp_domain.Exploration],
+        Tuple[str, Exception]
+    ]:
+        """Migrates exploration and transform exploration model into
+        exploration object.
+
+        Args:
+            exp_id: str. The ID of the exploration.
+            exp_model: ExplorationModel. The exploration model to migrate.
+
+        Returns:
+            Result((str, Exploration), (str, Exception)). Result containing
+            tuple that consists of exploration ID and either Exploration object
+            or Exception. Exploration object is returned when the migration was
+            successful and Exception is returned otherwise.
+        """
+        try:
+            exploration = exp_fetchers.get_exploration_from_model(exp_model)
+            exploration.validate()
+
+            with datastore_services.get_ndb_context():
+                if exp_services.get_story_id_linked_to_exploration(
+                        exp_id) is not None:
+                    exp_services.validate_exploration_for_story(
+                        exploration, True)
+
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((exp_id, e))
+        return result.Ok((exp_id, exploration))
+
+    @staticmethod
+    def _generate_exploration_changes(
+        exp_model: exp_models.ExplorationModel,
+        exploration: exp_domain.Exploration
+    ) -> Iterable[Tuple[str, exp_domain.ExplorationChange]]:
+        """Generates exploration change objects. The ExplorationChange object
+        is only generated when the exploration's states schema version is lower
+        than the latest schema version.
+
+        Args:
+            exp_model: ExplorationModel. The exploration model for which to
+                generate the change objects.
+            exploration: Exploration. The exploration domain object for which
+                to generate the change objects.
+
+        Yields:
+            (str, ExplorationChange). Tuple containing exploration ID and
+            ExplorationChange object.
+        """
+        exp_version = exp_model.version
+        if exp_version < exp_domain.Exploration.CURRENT_EXP_SCHEMA_VERSION:
```

Comment:
This code is wrong, I think we should generate the exp_change everytime.

Reply:
edited the condition. 
also, we might not need to generate exp_change everytime, because, if the exploration already has the attribute, then we don't have to commit exp_change again. wdyt ?